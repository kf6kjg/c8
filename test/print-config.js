/* global describe, before, beforeEach, it */

const { rm } = require('fs')
const { runSpawn, cleanJson, textGetConfigKey } = require('./print-config-helpers')
const c8Path = require.resolve('../bin/c8')
const nodePath = process.execPath
const chaiJestSnapshot = require('chai-jest-snapshot')
const pwd = process.cwd()
const { should } = require('chai')

// Todo: when bug mentioned below is fixed uncomment
// const { expect, should } = require('chai')

require('chai').should()
require('chai')
  .use(chaiJestSnapshot)
  .should()

before(cb => rm('tmp', { recursive: true, force: true }, cb))

beforeEach(function () {
  chaiJestSnapshot.configureUsingMochaContext(this)
})

describe('Help Message', () => {
  // Ensure the help message is correct - Snapshot of help message
  /**
   *   Test: Ensure Help Message is Correct
   *   Command: c8 --help
   *
   *   Runs the hep option and compares to a snapshot
   */
  it('ensure the help message is correct', () => {
    const output = runSpawn([c8Path, '--help'])
      .replaceAll(pwd, '.')
      .replaceAll(' ', '')
    output.should.matchSnapshot()
  })

  /**
   *  Test: Ensure 'not enough non-option arguments' warning message
   *  Command: c8
   *
   *  Runs c8 with incorrect options to make sure it produces a warning
   *
   */
  it('ensure \'not enough non-option arguments\' warning message', () => {
    const output = runSpawn([c8Path])
      .replaceAll(pwd, '.')
      .replaceAll(' ', '')
    output.should.matchSnapshot()
  })
})

describe('Print derived configuration CLI option', () => {
  // if print-config is true, c8 shouldn't demand any arguments
  /**
   *
   *  Test: should not demand any arguments: --print-config=true
   *  Command: c8 --print-config=true
   *
   *  if print-config is true, c8 shouldn't demand any arguments
   *
   */
  it('should not demand any arguments: --print-config=true', () => {
    const out = runSpawn([c8Path, '--print-config=true']).replaceAll(pwd, '.')
    out.should.matchSnapshot()
  })

  /**
   *
   *  Test: should not demand any arguments: --print-config
   *  Command: c8  --print-config
   *
   *  Other variation of if print-config is true, c8 shouldn't
   *  demand any arguments
   *
   */
  it('should not demand any arguments: --print-config', () => {
    const out = runSpawn([c8Path, '--print-config']).replaceAll(pwd, '.')
    out.should.matchSnapshot()
  })

  /**
   *
   *  Test: should demand arguments: --print-config=false
   *  Command: c8 --print-config=false
   *
   */
  it('should demand arguments: --print-config=false', () => {
    // Run Command./bin/c8.js --print-config=false
    const out = runSpawn([c8Path, '--print-config=false'])
      .replaceAll(pwd, '.')
      .replaceAll(' ', '')

    out.should.matchSnapshot()
  })

  /**
   *
   *  Test: should not demand arguments: --print-config=false
   *  Command: c8 --print-config=false --temp-directory=tmp/vanilla-all \
   *   --clean=false --all=true --include=test/fixtures/all/vanilla/**\/*.js
   *   --exclude=**\/*.ts node ./fixtures/all/vanilla/main
   *
   */
  it('should not demand arguments: --print-config=false', () => {
    const args = [
      c8Path,
      '--print-config=false',
      '--temp-directory=tmp/vanilla-all',
      '--clean=false',
      '--all=true',
      '--include=test/fixtures/all/vanilla/**/*.js',
      '--exclude=**/*.ts', // add an exclude to avoid default excludes of test/**
      nodePath,
      require.resolve('./fixtures/all/vanilla/main')
    ]
    const out = runSpawn(args).replaceAll(pwd, '.')
    out.should.matchSnapshot()
  })

  /**
   *
   *  Test: Ensure Valid JSON
   *  Command: c8 --print-config --print-config-format=json
   *
   *  ensure --print-config-format=json prints valid json document
   */
  it('ensure valid json', () => {
    const out = runSpawn([c8Path, '--print-config', '--print-config-format=json']).replaceAll(pwd, '.')
    try {
      const jOut = cleanJson(out)
      jOut.should.matchSnapshot()
    } catch (e) {
      should.fail('invalid json document produced from --print-config option')
    }
  })

  /**
   *
   *  Test: Ensure comma delimited values transform into an array
   *  Command: C8 --reporter=lcov,text --print-config --print-config-format=json
   *
   *  Todo:  There is a bug in yargs where this is not transformedd into an array
   *   Skipping test for now
   */
  it('ensure comma delimited values transform into an array', function () {
    this.skip()
    const out = runSpawn([
      c8Path,
      '--reporter=lcov,text',
      '--print-config',
      '--print-config-format=json'
    ]).replaceAll(pwd, '.')

    const jOut = cleanJson(out)

    const includesKey = Object.keys(jOut).includes('reporter')
    const checkFor = ['lcov', 'text']
    includesKey.should.eql(true)
    jOut.reporter.should.eql(checkFor)
  })

  /**
   *
   *  Test: Ensure default project configuration file is loaded
   *  Command: c8  --print-config
   *
   */
  it('ensure default project configuration file is loaded', () => {
    const out = runSpawn([c8Path, '--print-config', '--print-config-format=json']).replaceAll(pwd, '.')
    const jOut = cleanJson(out)

    const includesKey = Object.keys(jOut).includes('config')
    includesKey.should.eql(true)
    jOut.config.should.eql('./.nycrc')
  })

  ;['text', 'json'].forEach((format) => {
    describe(`${format} format option`, () => {
      /**
       *
       *  Test: ensure loads config file from cli
       *  Command: c8 -c ./test/fixtures/config/.c8rc.json --print-config --print-config-format=json|text
       *
       */
      it('ensure loads config file from cli', () => {
        const out = runSpawn([
          c8Path,
          '--config=./test/fixtures/config/.c8rc.json',
          '--print-config',
          `--print-config-format=${format}`
        ]).replaceAll(pwd, '.')

        if (format === 'json') {
          const jOut = cleanJson(out)

          Object.keys(jOut).includes('config').should.eql(true)
          jOut.config.should.eql('./test/fixtures/config/.c8rc.json')
        } else if (format === 'text') {
          const value = textGetConfigKey(out, 'config')
          if (value) {
            value.should.eql('./test/fixtures/config/.c8rc.json')
          } else {
            should.fail('couldn\'t find configuration value for option --config')
          }
        }
      })

      /**
       *
       *  Test: Ensure loads reporter option from cli
       *  Command: c8 --reporter=lcov --print-config
       *
       */
      it('ensure loads reporter option from cli', () => {
        const out = runSpawn([
          c8Path,
          '--reporter=lcov',
          '--print-config',
          `--print-config-format=${format}`
        ]).replaceAll(pwd, '.')

        if (format === 'json') {
          const jOut = cleanJson(out)

          Object.keys(jOut).includes('reporter').should.eql(true)
          jOut.reporter.should.eql('lcov')
        } else if (format === 'text') {
          const value = textGetConfigKey(out, 'reporter')
          if (value) {
            // Todo: when the load comma delimited text array bug is fixed, need to adjust this line
            value.should.eql('lcov')
          } else {
            should.fail('couldn\'t find configuration value for option --reporter')
          }
        }
      })
    })
  })
})
