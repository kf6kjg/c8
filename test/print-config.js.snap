// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help Message ensure 'not enough non-option arguments' warning message 1`] = `",,c8.js[opts][script][opts]Commands:c8.jscheck-coveragecheckwhethercoverageiswithinthresholdsprovidedc8.jsreportreadV8coveragedatafromtempandoutputreportReportingoptions-r,--reportercoveragereporter(s)touse[default:\\"text\\"]-o,--reports-dir,--report-dirdirectorywherecoveragereportswillbeoutputto[default:\\"./coverage\\"]--allsupplying--allwillcausec8toconsiderallsrcfilesinthecurrentworkingdirectorywhenthedeterminingcoverage.Respectsinclude/exclude.[boolean][default:false]--srcsupplying--srcwilloverridecwdasthedefaultlocationwhere--alllooksforsrcfiles.--srccanbesuppliedmultipletimesandeachdirectorywillbeincluded.Thisallowsforworkspacesspanningmultipleprojects[string]-n,--includealistofspecificfilesthatshouldbecovered(globpatternsaresupported)[default:[]]-x,--excludealistofspecificfilesanddirectoriesthatshouldbeexcludedfromcoverage(globpatternsaresupported)[default:[\\"coverage/**\\",\\"packages/*/test{,s}/**\\",\\"**/*.d.ts\\",\\"test{,s}/**\\",\\"test{,-*}.{js,cjs,mjs,ts,tsx,jsx}\\",\\"**/*{.,-}test.{js,cjs,mjs,ts,tsx,jsx}\\",\\"**/__tests__/**\\",\\"**/{ava,babel,nyc}.config.{js,cjs,mjs}\\",\\"**/jest.config.{js,cjs,mjs,ts}\\",\\"**/{karma,rollup,webpack}.config.js\\",\\"**/.{eslint,mocha}rc.{js,cjs}\\"]]-e,--extensionalistofspecificfileextensionsthatshouldbecovered[default:[\\".js\\",\\".cjs\\",\\".mjs\\",\\".ts\\",\\".tsx\\",\\".jsx\\"]]-a,--exclude-after-remapapplyexcludelogictofilesaftertheyareremappedbyasource-map[boolean][default:false]--skip-fulldonotshowfileswith100%statement,branch,andfunctioncoverage[boolean][default:false]Coveragethresholds--check-coveragecheckwhethercoverageiswithinthresholdsprovided[boolean][default:false]--brancheswhat%ofbranchesmustbecovered?[number][default:0]--functionswhat%offunctionsmustbecovered?[number][default:0]--lineswhat%oflinesmustbecovered?[number][default:90]--statementswhat%ofstatementsmustbecovered?[number][default:0]--per-filecheckthresholdsperfile[boolean][default:false]--100shortcutfor--check-coverage--lines100--functions100--branches100--statements100[boolean][default:false]Options:--helpShowhelp[boolean]--versionShowversionnumber[boolean]-c,--configpathtoJSONconfigurationfile[default:(default)]--exclude-node-moduleswhetherornottoexcludeallnode_modulefolders(i.e.**/node_modules/**)bydefault[boolean][default:true]--temp-directorydirectoryV8coveragedataiswrittentoandreadfrom[default:\\"C:Users15712OneDriveDesktopc8coveragetmp\\"]--cleanshouldtempfilesbedeletedbeforescriptexecution[boolean][default:true]--resolveresolvepathstoalternatebasedirectory[default:\\"\\"]--wrapper-lengthhowmanybytesisthewrapperprefixonexecutedJavaScript[number]--omit-relativeomitanypathsthatarenotabsolute,e.g.,internal/net.js[boolean][default:true]--allowExternalsupplying--allowExternalwillcausec8toallowfilesfromoutsideofyourcwd.Thisappliesbothtofilesdiscoveredincoveragetempfilesandalsosrcfilesdiscoveredifusingthe--allflag.[boolean][default:false]--merge-asyncsupplying--merge-asyncwillmergeallv8coveragereportsasynchronouslyandincrementally.ThisistoavoidOOMissueswithNode.jsruntime.[boolean][default:false]--print-configPrintthederivedconfigurationbetweencommandlineparametersandloadedconfigurationfile[boolean][default:false]--print-config-formatFormattoprinttheconfigurationin.Acceptedformatsareeithertextorjson[string][default:\\"text\\"]visithttps://git.io/vHysAforlistofavailablereportersNotenoughnon-optionarguments:got0,needatleast1"`;

exports[`Help Message ensure the help message is correct 1`] = `",c8.js[opts][script][opts]Commands:c8.jscheck-coveragecheckwhethercoverageiswithinthresholdsprovidedc8.jsreportreadV8coveragedatafromtempandoutputreportReportingoptions-r,--reportercoveragereporter(s)touse[default:\\"text\\"]-o,--reports-dir,--report-dirdirectorywherecoveragereportswillbeoutputto[default:\\"./coverage\\"]--allsupplying--allwillcausec8toconsiderallsrcfilesinthecurrentworkingdirectorywhenthedeterminingcoverage.Respectsinclude/exclude.[boolean][default:false]--srcsupplying--srcwilloverridecwdasthedefaultlocationwhere--alllooksforsrcfiles.--srccanbesuppliedmultipletimesandeachdirectorywillbeincluded.Thisallowsforworkspacesspanningmultipleprojects[string]-n,--includealistofspecificfilesthatshouldbecovered(globpatternsaresupported)[default:[]]-x,--excludealistofspecificfilesanddirectoriesthatshouldbeexcludedfromcoverage(globpatternsaresupported)[default:[\\"coverage/**\\",\\"packages/*/test{,s}/**\\",\\"**/*.d.ts\\",\\"test{,s}/**\\",\\"test{,-*}.{js,cjs,mjs,ts,tsx,jsx}\\",\\"**/*{.,-}test.{js,cjs,mjs,ts,tsx,jsx}\\",\\"**/__tests__/**\\",\\"**/{ava,babel,nyc}.config.{js,cjs,mjs}\\",\\"**/jest.config.{js,cjs,mjs,ts}\\",\\"**/{karma,rollup,webpack}.config.js\\",\\"**/.{eslint,mocha}rc.{js,cjs}\\"]]-e,--extensionalistofspecificfileextensionsthatshouldbecovered[default:[\\".js\\",\\".cjs\\",\\".mjs\\",\\".ts\\",\\".tsx\\",\\".jsx\\"]]-a,--exclude-after-remapapplyexcludelogictofilesaftertheyareremappedbyasource-map[boolean][default:false]--skip-fulldonotshowfileswith100%statement,branch,andfunctioncoverage[boolean][default:false]Coveragethresholds--check-coveragecheckwhethercoverageiswithinthresholdsprovided[boolean][default:false]--brancheswhat%ofbranchesmustbecovered?[number][default:0]--functionswhat%offunctionsmustbecovered?[number][default:0]--lineswhat%oflinesmustbecovered?[number][default:90]--statementswhat%ofstatementsmustbecovered?[number][default:0]--per-filecheckthresholdsperfile[boolean][default:false]--100shortcutfor--check-coverage--lines100--functions100--branches100--statements100[boolean][default:false]Options:--helpShowhelp[boolean]--versionShowversionnumber[boolean]-c,--configpathtoJSONconfigurationfile[default:(default)]--exclude-node-moduleswhetherornottoexcludeallnode_modulefolders(i.e.**/node_modules/**)bydefault[boolean][default:true]--temp-directorydirectoryV8coveragedataiswrittentoandreadfrom[default:\\"C:Users15712OneDriveDesktopc8coveragetmp\\"]--cleanshouldtempfilesbedeletedbeforescriptexecution[boolean][default:true]--resolveresolvepathstoalternatebasedirectory[default:\\"\\"]--wrapper-lengthhowmanybytesisthewrapperprefixonexecutedJavaScript[number]--omit-relativeomitanypathsthatarenotabsolute,e.g.,internal/net.js[boolean][default:true]--allowExternalsupplying--allowExternalwillcausec8toallowfilesfromoutsideofyourcwd.Thisappliesbothtofilesdiscoveredincoveragetempfilesandalsosrcfilesdiscoveredifusingthe--allflag.[boolean][default:false]--merge-asyncsupplying--merge-asyncwillmergeallv8coveragereportsasynchronouslyandincrementally.ThisistoavoidOOMissueswithNode.jsruntime.[boolean][default:false]--print-configPrintthederivedconfigurationbetweencommandlineparametersandloadedconfigurationfile[boolean][default:false]--print-config-formatFormattoprinttheconfigurationin.Acceptedformatsareeithertextorjson[string][default:\\"text\\"]visithttps://git.io/vHysAforlistofavailablereporters,"`;

exports[`Print derived configuration CLI option ensure valid json 1`] = `
Object {
  "100": false,
  "all": false,
  "allow-external": false,
  "branches": 82,
  "check-coverage": false,
  "clean": true,
  "config": "C:\\\\Users\\\\15712\\\\OneDrive\\\\Desktop\\\\c8\\\\.nycrc",
  "exclude": Array [
    "coverage/**",
    "packages/*/test{,s}/**",
    "**/*.d.ts",
    "test{,s}/**",
    "test{,-*}.{js,cjs,mjs,ts,tsx,jsx}",
    "**/*{.,-}test.{js,cjs,mjs,ts,tsx,jsx}",
    "**/__tests__/**",
    "**/{ava,babel,nyc}.config.{js,cjs,mjs}",
    "**/jest.config.{js,cjs,mjs,ts}",
    "**/{karma,rollup,webpack}.config.js",
    "**/.{eslint,mocha}rc.{js,cjs}",
  ],
  "exclude-after-remap": false,
  "exclude-node-modules": true,
  "extension": Array [
    ".js",
    ".cjs",
    ".mjs",
    ".ts",
    ".tsx",
    ".jsx",
  ],
  "functions": 0,
  "include": Array [],
  "lines": 95,
  "merge-async": false,
  "omit-relative": true,
  "per-file": false,
  "print-config": true,
  "print-config-format": "json",
  "report-dir": "./coverage",
  "reporter": Array [
    "html",
    "text",
  ],
  "reports-dir": "./coverage",
  "resolve": "",
  "skip-full": false,
  "statements": 95,
  "temp-directory": "C:\\\\Users\\\\15712\\\\OneDrive\\\\Desktop\\\\c8\\\\coverage\\\\tmp",
}
`;

exports[`Print derived configuration CLI option should demand arguments: --print-config=false 1`] = `",,c8.js[opts][script][opts]Commands:c8.jscheck-coveragecheckwhethercoverageiswithinthresholdsprovidedc8.jsreportreadV8coveragedatafromtempandoutputreportReportingoptions-r,--reportercoveragereporter(s)touse[default:\\"text\\"]-o,--reports-dir,--report-dirdirectorywherecoveragereportswillbeoutputto[default:\\"./coverage\\"]--allsupplying--allwillcausec8toconsiderallsrcfilesinthecurrentworkingdirectorywhenthedeterminingcoverage.Respectsinclude/exclude.[boolean][default:false]--srcsupplying--srcwilloverridecwdasthedefaultlocationwhere--alllooksforsrcfiles.--srccanbesuppliedmultipletimesandeachdirectorywillbeincluded.Thisallowsforworkspacesspanningmultipleprojects[string]-n,--includealistofspecificfilesthatshouldbecovered(globpatternsaresupported)[default:[]]-x,--excludealistofspecificfilesanddirectoriesthatshouldbeexcludedfromcoverage(globpatternsaresupported)[default:[\\"coverage/**\\",\\"packages/*/test{,s}/**\\",\\"**/*.d.ts\\",\\"test{,s}/**\\",\\"test{,-*}.{js,cjs,mjs,ts,tsx,jsx}\\",\\"**/*{.,-}test.{js,cjs,mjs,ts,tsx,jsx}\\",\\"**/__tests__/**\\",\\"**/{ava,babel,nyc}.config.{js,cjs,mjs}\\",\\"**/jest.config.{js,cjs,mjs,ts}\\",\\"**/{karma,rollup,webpack}.config.js\\",\\"**/.{eslint,mocha}rc.{js,cjs}\\"]]-e,--extensionalistofspecificfileextensionsthatshouldbecovered[default:[\\".js\\",\\".cjs\\",\\".mjs\\",\\".ts\\",\\".tsx\\",\\".jsx\\"]]-a,--exclude-after-remapapplyexcludelogictofilesaftertheyareremappedbyasource-map[boolean][default:false]--skip-fulldonotshowfileswith100%statement,branch,andfunctioncoverage[boolean][default:false]Coveragethresholds--check-coveragecheckwhethercoverageiswithinthresholdsprovided[boolean][default:false]--brancheswhat%ofbranchesmustbecovered?[number][default:0]--functionswhat%offunctionsmustbecovered?[number][default:0]--lineswhat%oflinesmustbecovered?[number][default:90]--statementswhat%ofstatementsmustbecovered?[number][default:0]--per-filecheckthresholdsperfile[boolean][default:false]--100shortcutfor--check-coverage--lines100--functions100--branches100--statements100[boolean][default:false]Options:--helpShowhelp[boolean]--versionShowversionnumber[boolean]-c,--configpathtoJSONconfigurationfile[default:(default)]--exclude-node-moduleswhetherornottoexcludeallnode_modulefolders(i.e.**/node_modules/**)bydefault[boolean][default:true]--temp-directorydirectoryV8coveragedataiswrittentoandreadfrom[default:\\"C:Users15712OneDriveDesktopc8coveragetmp\\"]--cleanshouldtempfilesbedeletedbeforescriptexecution[boolean][default:true]--resolveresolvepathstoalternatebasedirectory[default:\\"\\"]--wrapper-lengthhowmanybytesisthewrapperprefixonexecutedJavaScript[number]--omit-relativeomitanypathsthatarenotabsolute,e.g.,internal/net.js[boolean][default:true]--allowExternalsupplying--allowExternalwillcausec8toallowfilesfromoutsideofyourcwd.Thisappliesbothtofilesdiscoveredincoveragetempfilesandalsosrcfilesdiscoveredifusingthe--allflag.[boolean][default:false]--merge-asyncsupplying--merge-asyncwillmergeallv8coveragereportsasynchronouslyandincrementally.ThisistoavoidOOMissueswithNode.jsruntime.[boolean][default:false]--print-configPrintthederivedconfigurationbetweencommandlineparametersandloadedconfigurationfile[boolean][default:false]--print-config-formatFormattoprinttheconfigurationin.Acceptedformatsareeithertextorjson[string][default:\\"text\\"]visithttps://git.io/vHysAforlistofavailablereportersNotenoughnon-optionarguments:got0,needatleast1"`;

exports[`Print derived configuration CLI option should not demand any arguments: --print-config 1`] = `",/*________/______/____*//*_____/////////_____////////__*//*___//_____________/_____/__*//*__/_______________////___*//*_/________________////////__*//*_//______________/______//_*//*__///___________//______/__*//*____////___////___*//*_______/////////______/////////_____*/CommandIssued:c8--print-configConfigFileLoaded:C:Users15712OneDriveDesktopc8.nycrcDerivedConfigurationfromCLIoptionsandconfigurationfile------------------------------------------------------------------------------100:falseprint-config:truereporter:[\\"html\\",\\"text\\"]lines:95branches:82statements:95config:'C:Users15712OneDriveDesktopc8.nycrc'reports-dir:'./coverage'report-dir:'./coverage'all:falsesrc:undefinedexclude-node-modules:trueinclude:exclude:[\\"coverage/**\\",\\"packages/*/test{,s}/**\\",\\"**/*.d.ts\\",\\"test{,s}/**\\",\\"test{,-*}.{js,cjs,mjs,ts,tsx,jsx}\\",\\"**/*{.,-}test.{js,cjs,mjs,ts,tsx,jsx}\\",\\"**/__tests__/**\\",\\"**/{ava,babel,nyc}.config.{js,cjs,mjs}\\",\\"**/jest.config.{js,cjs,mjs,ts}\\",\\"**/{karma,rollup,webpack}.config.js\\",\\"**/.{eslint,mocha}rc.{js,cjs}\\"]extension:[\\".js\\",\\".cjs\\",\\".mjs\\",\\".ts\\",\\".tsx\\",\\".jsx\\"]exclude-after-remap:falseskip-full:falsecheck-coverage:falsefunctions:0per-file:falsetemp-directory:'C:Users15712OneDriveDesktopc8coveragetmp'clean:trueresolve:''omit-relative:trueallow-external:falsemerge-async:falseprint-config-format:'text',"`;

exports[`Print derived configuration CLI option should not demand any arguments: --print-config=true 1`] = `",/*________/______/____*//*_____/////////_____////////__*//*___//_____________/_____/__*//*__/_______________////___*//*_/________________////////__*//*_//______________/______//_*//*__///___________//______/__*//*____////___////___*//*_______/////////______/////////_____*/CommandIssued:c8--print-config=trueConfigFileLoaded:C:Users15712OneDriveDesktopc8.nycrcDerivedConfigurationfromCLIoptionsandconfigurationfile------------------------------------------------------------------------------100:falseprint-config:truereporter:[\\"html\\",\\"text\\"]lines:95branches:82statements:95config:'C:Users15712OneDriveDesktopc8.nycrc'reports-dir:'./coverage'report-dir:'./coverage'all:falsesrc:undefinedexclude-node-modules:trueinclude:exclude:[\\"coverage/**\\",\\"packages/*/test{,s}/**\\",\\"**/*.d.ts\\",\\"test{,s}/**\\",\\"test{,-*}.{js,cjs,mjs,ts,tsx,jsx}\\",\\"**/*{.,-}test.{js,cjs,mjs,ts,tsx,jsx}\\",\\"**/__tests__/**\\",\\"**/{ava,babel,nyc}.config.{js,cjs,mjs}\\",\\"**/jest.config.{js,cjs,mjs,ts}\\",\\"**/{karma,rollup,webpack}.config.js\\",\\"**/.{eslint,mocha}rc.{js,cjs}\\"]extension:[\\".js\\",\\".cjs\\",\\".mjs\\",\\".ts\\",\\".tsx\\",\\".jsx\\"]exclude-after-remap:falseskip-full:falsecheck-coverage:falsefunctions:0per-file:falsetemp-directory:'C:Users15712OneDriveDesktopc8coveragetmp'clean:trueresolve:''omit-relative:trueallow-external:falsemerge-async:falseprint-config-format:'text',"`;

exports[`Print derived configuration CLI option should not demand arguments: --print-config=false 1`] = `",zeropositivenegative--------------|---------|----------|---------|---------|-------------------File|%Stmts|%Branch|%Funcs|%Lines|UncoveredLine#s--------------|---------|----------|---------|---------|-------------------Allfiles|64.28|66.66|50|64.28|vanilla|78.26|75|100|78.26|loaded.js|73.68|71.42|100|73.68|4-5,16-18main.js|100|100|100|100|vanilla/dir|0|0|0|0|unloaded.js|0|0|0|0|1-5--------------|---------|----------|---------|---------|-------------------,"`;
